pkg/operator/certrotationcontroller/certrotationcontroller.go

```diff
@@ -812,10 +812,10 @@ func newCertRotationController(
 				AutoRegenerateAfterOfflineExpiry: "https://github.com/openshift/cluster-kube-apiserver-operator/pull/1631,'[Conformance][sig-api-machinery][Feature:APIServer] local kubeconfig \"localhost-recovery.kubeconfig\" should be present on all masters and work [apigroup:config.openshift.io] [Suite:openshift/conformance/parallel/minimal]'",
 				Description:                      "Signer for the per-master-debugging-client.",
 			},
-			Validity: devRotationExceptionYear,
+			Validity: 3 * devRotationExceptionYear,
 			// Refresh set to 80% of the validity.
 			// This range is consistent with most other signers defined in this pkg.
-			Refresh:                devRotationExceptionTenMonth,
+			Refresh:                3 * devRotationExceptionTenMonth,
```

Comment:
since we have a local controller running on each machine for certificate recovery, how is it possible that this certificate/signer is not being regenerated?

Reply:
We're running certrotation controller in RefreshOnlyWhenExpired mode to update node-system-admin-client secret, but we're not running nodekubeconfigcontroller to generate new kubeconfigs on disk. 

Should we run it - yes, maybe, after all we already run resourcesync controller after all. However, it would make recovery procedure more unpredictable - i.e. the controller may delay kubeconfig generation and users will attempt to approve CSR with expired kubeconfig. Also, it will allow customers use it in unsupported configurations - i.e. after the cluster was shutdown for 3 years - I don't know the implications of that yet